app/javascript/components/progress-graph/ProgressGraph.tsx

```diff
@@ -0,0 +1,87 @@
+import * as React from 'react'
+import { useRef } from 'react'
+
+import {
+  minMaxNormalize,
+  drawSegmentPath,
+  drawSmoothPath,
+  DataPoint,
+} from './svg-graph-util'
+
+interface IProgressGraph {
+  data: Array<number>
+  height: number
+  width: number
+  smooth?: boolean
+}
+
+export const ProgressGraph: React.FC<IProgressGraph> = ({
+  data,
+  height,
+  width,
+  smooth = false,
+}) => {
+  const randomIdRef = useRef<null | number>(null)
+
+  function getRandomId() {
+    const currentId = randomIdRef.current
+    if (currentId !== null) {
+      return currentId
+    }
+    const newId = Math.floor(Math.random() * 10000)
+    randomIdRef.current = newId
+    return newId
+  }
```

Comment:
generates a random number to append to the id of the gradient to prevent collision and the id being overridden by multiple graphs appearing on one page